java/clients/src/test/java/sleeper/clients/QueryWebSocketClientTest.java

```diff
@@ -299,7 +300,8 @@ void shouldHandleMalformedJson() throws Exception {
             // When / Then
             assertThat(runQueryFuture(query,
                     withResponses(
-                            message("{"))))
+                            message("{"),
+                            close("Connection closed normally"))))
```

Comment:
Is it a bug that the connection isn't closed automatically when there's malformed JSON? Also note it won't be closed normally, it'll time out.

Reply:
Thanks, this does look like a bug since the connection closed last time when message validation failed. This was causing the base PR to hang when running queries so I've fixed this in the base PR